### ISSUE TITLE
`rootfs-builder` omits `/usr/lib/gcc` directory from FHSEnv

### ISSUE THREAD

**Issue by timschumi** [NixOS/nixpkgs#365267]
## Describe the bug

Starting with commit a182a53243828257018c6ef4c7d2edbe89fd63e5 (the introduction of `rootfs-builder`), the resulting FHSEnv no longer contains the `/usr/lib/gcc` directory, even if a package with a `lib/gcc` directory is included (e.g. `libgcc`).

I can't immediately spot why this is the case (a matching mapping seems like it would get created).

## Steps To Reproduce

Evaluate the following FHSEnv:

```
pkgs.buildFHSEnv {
  name = "test";
  multiPkgs = pkgs: (with pkgs; [
    libgcc
  ]);
}
```

Observe that `/usr/lib/gcc` is missing.

## Expected behavior

`/usr/lib/gcc` should be present.

## Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.63, NixOS, 25.05 (Warbler), 25.05.20241205.d0797a0`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - nixpkgs: `/nix/store/nr5nl3zwzl02x3rnikjbry3s5xy7bm1d-source`

## Notify maintainers

No dedicated `meta.maintainers` for this, tagging the person who is the author of `rootfs-builder`.

@K900

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by K900** [comment]
`libgcc` does not contain `lib/gcc` though? It only contains `lib/libgcc_s.so.1`.

**Comment by timschumi** [comment]
Interesting, `nix-locate` output (at least in my understanding) seems to indicate that it's directly contained, although I have seen by now that it actually specifies a `gcc` path when asked:

```
$ nix-locate -w crtendS.o
[...]
libgcc.out                                        1,160 r /nix/store/xzfmarrq8x8s4ivpya24rrndqsq2ndiz-gcc-13.3.0/lib/gcc/x86_64-unknown-linux-gnu/13.3.0/crtendS.o
[...]
```

For what it's worth, replacing `libgcc` with `gcc` in the (now edited) reproducer yields the exact same result.
The only package that works as "expected" is `gcc-unwrapped`.

**Comment by K900** [comment]
That's just a nix-locate thing, it has weird interactions with aliases. `libgcc.out` will probably work.

**Comment by timschumi** [comment]
This is probably getting a bit off-topic, but is there a notable difference between `libgcc` and `libgcc.out` in this context? According to the [stdenv definition](https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/check-meta.nix#L464-L482) and [the documentation](https://nixos.org/manual/nixpkgs/unstable/#chap-multiple-output) the installed outputs should default to `out` and `man`, given that `bin` is not available on `libgcc`.

As for this issue in particular, I suppose it has shifted from "broken behavior" to "undocumented behavior change"

**Comment by timschumi** [comment]
Of course, a minute after posting I figure out that `libgcc` is really just `gcc.libgcc`, which is why it does not default to `.out`.

Still, it seems like there is a behavior change to be documented here.
